strawberry/utils/dataclasses.py

```diff
@@ -0,0 +1,91 @@
+import sys
+from dataclasses import (  # type: ignore
+    _FIELD,
+    _FIELD_INITVAR,
+    _FIELDS,
+    _HAS_DEFAULT_FACTORY,
+    _POST_INIT_NAME,
+    MISSING,
+    _create_fn,
+    _field_init,
+    _init_param,
+    _set_new_attribute,
+)
+
+
+def dataclass_init_fn(fields, frozen, has_post_init, self_name, globals):
+    """
+    We create a custom __init__ function for the dataclasses that back
+    Strawberry object types to only accept keyword arguments. This allows us to
+    avoid the problem where a type cannot define a field with a default value
+    before a field that doesn't have a default value.
+
+    An example of the problem:
+    https://stackoverflow.com/questions/51575931/class-inheritance-in-python-3-7-dataclasses
+
+    Code is adapted from:
+    https://github.com/python/cpython/blob/v3.9.6/Lib/dataclasses.py#L489-L536
```

Comment:
Not sure how to include the license. @BryceBeagle any suggestions?

Reply:
I think the best method would be to extract the stuff from cpython as best as we can to it's own directory+file and then place the cpython license in that directory. Then add a comment to the top of the file saying the code in the file abides by a different license. That way there's no confusion to which files the license applies to, and on the flip-side what license applies to that file.